app/controllers/actions/IdentifierAction.scala

```diff
@@ -34,32 +37,43 @@ trait IdentifierAction extends ActionBuilder[IdentifierRequest, AnyContent] with
 class AuthenticatedIdentifierAction @Inject()(
   override val authConnector: AuthConnector,
   config: FrontendAppConfig,
-  val parser: BodyParsers.Default
+  val parser: BodyParsers.Default,
+  enrolmentStoreConnector: EnrolmentStoreConnector
 )(implicit val executionContext: ExecutionContext)
     extends IdentifierAction
     with AuthorisedFunctions {
 
-  private val enrolmentIdentifierKey: String = "VATRegNoTURN"
-
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter
       .fromHeadersAndSession(request.headers, Some(request.session))
 
-    authorised(Enrolment(config.enrolmentKey))
-      .retrieve(Retrievals.authorisedEnrolments) {
-        enrolments =>
-          val eoriNumber: String = (for {
-            enrolment  <- enrolments.enrolments.find(_.key.equals(config.enrolmentKey))
-            identifier <- enrolment.getIdentifier(enrolmentIdentifierKey)
-          } yield identifier.value).getOrElse(throw InsufficientEnrolments(s"Unable to retrieve enrolment for $enrolmentIdentifierKey"))
-
-          block(IdentifierRequest(request, eoriNumber))
+    authorised(EmptyPredicate)
+      .retrieve(Retrievals.allEnrolments and Retrievals.groupIdentifier) {
+        case enrolments ~ maybeGroupId =>
+          (for {
+            enrolment <- enrolments.enrolments.filter(_.isActivated).find(_.key.equals(config.enrolmentKey))
+          } yield
+            enrolment.getIdentifier(config.enrolmentIdentifierKey) match {
+              case Some(eoriNumber) => block(IdentifierRequest(request, eoriNumber.value))
+              case _                => Future.successful(Redirect(routes.UnauthorisedController.onPageLoad()))
+            }).getOrElse(checkForGroupEnrolment(maybeGroupId))
       }
   } recover {
     case _: NoActiveSession =>
       Redirect(config.loginUrl, Map("continue" -> Seq(config.loginContinueUrl)))
     case _: AuthorisationException =>
       Redirect(routes.UnauthorisedController.onPageLoad())
   }
+
+  private def checkForGroupEnrolment(maybeGroupId: Option[String])(implicit hc: HeaderCarrier): Future[Result] =
+    maybeGroupId match {
+      case Some(groupId) =>
+        enrolmentStoreConnector.checkGroupEnrolments(groupId, config.enrolmentKey) map {
+          case true  => Redirect(routes.UnauthorisedWithGroupAccessController.onPageLoad())
+          case false => Redirect(routes.UnauthorisedWithoutGroupAccessController.onPageLoad())
+        }
```

Comment:
I don't think that we should have separate controllers for these, instead I think it would be better if we return a 401 here and the content.